### DISCUSSION TITLE
NextJs SSR, client/server fetches

### DISCUSSION THREAD

**Discussion by arkhnchul** [coinbase/rest-hooks#2911] [Q&A]
Hi all. Where the dataclient fetches should take place by default when used with NextJs - on the server or on the client side? I am checking the included nextjs example, and despite the "No fetch requests took place on the client" statement, i see the requests to api.exchange.coinbase.com in the browser console and a warnings "Fetch appears unhandled - you are likely missing the NetworkManager middleware" for each request.

**Comment by ntucker** [comment]
I went to https://stackblitz.com/github/reactive/data-client/tree/master/examples/nextjs?file=pages%2FAssetPrice.tsx and do not see anything in the dev console other than `Download the React DevTools for a better development experience: https://reactjs.org/link/react-devtools`. Where was it you went that you saw this console warning?

As far as where fetches happen - potentially both! The server does the fetch for initial page load, however, the page remains interactive, so in this case after 2 seconds it will fetch the current price to [update it continually using polling](https://dataclient.io/docs/api/useLive). Client-side fetches could also come due to [mutations](https://dataclient.io/docs/getting-started/mutations), etc. The point is just that there is no overfetching. There is no need for the client to refetch data it already has, as the store is able to hydrate with the server side load. *New* data still needs to fetch.

If the polling is too confusing, you can try replacing [useLive](https://dataclient.io/docs/api/useLive) with [useSuspense](https://dataclient.io/docs/api/useSuspense), which will not poll the price continually after load.

**Comment by ntucker** [comment]
I updated the [NextJS demo ](https://dataclient.io/demos) homepage to have an interactive todolist instead of the live price. This should make the lack of client-side fetching for SSR more obvious.
